packages/core/src/modules/non-secret/repository/NonSecretRecord.ts

```diff
@@ -0,0 +1,47 @@
+import type { RecordTags, TagsBase } from '../../../storage/BaseRecord'
+
+import { BaseRecord } from '../../../storage/BaseRecord'
+import { uuid } from '../../../utils/uuid'
+
+export type NonSecretRecordTags = TagsBase
+export type DefaultNonSecretRecordTags = {
+  connectionId: string
```

Comment:
I think the `connectionId` is slightly opinionated. I can imagine that a lot of use cases do not benefit from this. I do understand that it can be used for getting this record via a connection, but then why would you not use the `connectonRecord.metadata`?

Reply:
When I first saw this PR, I thought that I could use to store some wallet-specific preferences, which of course are not tied to any connection, so I agree that leveraging the use of metadata would make this module more generic.

Reply:
While I know that you can manually add the connectionId to the tg but remember most of the business relies on adding custom history messages that depends on interaction with some sort of a connection with other agent, so why not automate this and just have the connectionID as part of the tags after all, 
The other reason is when removing the connectionId, if found that the getTags function fails to comply with base class so its kind of glued to the framework.
For metadata I do not recommend shuffling much of metadata into the tags and let that be on the controller side to pickup from. I would rather make as simple as the id of the connection that had this data

Reply:
We've had multiple use cases where external data is not tied to a specific connection. For now we've always used the record metadata, but for larger objects I can understand you would use an external record. For this reason I think it is important to keep this general and not tied to any connection. Implementations that want to tie this to a connection can do some by storing the connectionId in the record payload.